nemo/collections/common/data/dataset.py

```diff
@@ -286,3 +289,310 @@ def __len__(self):
+            Defaults to 80
+        pause_end (int): terminates all CS samples with silence equal to this value (msecs)
+            Defaults to 20
+        sampling_scales (list or float): gives you the ability to upsample/downsample by language
+        seed: Optional value to seed the numpy RNG.
+        global_rank (int): Worker rank, used for partitioning map style datasets. Defaults to 0.
+        world_size (int): Total number of processes, used for partitioning map style datasets. Defaults to 1.
+        pure_random (bool): If true, then always draw random sample from lang_probs. If false, you only draw from those languages
+                            which you haven't sampled from yet for the composite sample
+        force_monochannel (bool): If true, then all output audio will be mono-channel
+        infinity_mode (bool): If true, then the dataset iterable will generate an infinite amount of samples
+        sample_rate (int): the sample rate of all audio being sent to this Dataset
+        augmentor (AudioAugmentor): The any perturbations you wish to have applied on the CS samples
+    """
+
+    def __init__(
+        self,
+        datasets: dict,
+        lang_probs: List[float] = None,
+        shuffle: bool = True,
+        min_duration: int = 16,
+        max_duration: int = 20,
+        min_monolingual: float = 0.2,
+        db_norm: float = -25.0,
+        pause_start: int = 20,
+        pause_join: int = 80,
+        pause_end: int = 20,
+        sampling_scales: Optional[Union[float, List[float]]] = None,
+        seed: Optional[int] = None,
+        global_rank: int = 0,
+        world_size: int = 1,
+        pure_random: bool = False,
+        force_monochannel: bool = False,
+        infinity_mode: bool = False,
+        sample_rate: int = 16000,
+        augmentor: Optional['AudioAugmentor'] = None,
+    ):
+        super().__init__()
+
+        self.datasets = datasets
+        self.langs = list(datasets.keys())
+        self.langs_set = set(self.langs)
+        self.lang_iterables = {k: None for k in self.langs}
+        self.lang_kind = {k: None for k in self.langs}
+        self.shuffle = shuffle
+        self.min_duration = min_duration
+        self.max_duration = max_duration
+        self.min_monolingual = min_monolingual
+        self.db_norm = db_norm
+        self.pause_start = pause_start
+        self.pause_join = pause_join
+        self.pause_end = pause_end
+        self.pure_random = pure_random
+        self.force_monochannel = force_monochannel
+        self.infinity_mode = infinity_mode
+        self.global_rank = global_rank
+        self.world_size = world_size
+        self.augmentor = augmentor
+        self.sample_rate = sample_rate
+        self.length = 0
+        if lang_probs is None:
+            self.prob_dict = {l: 1.0 / len(self.langs) for l in self.langs}
+        else:
+            self.prob_dict = {l: lang_probs[i] for i, l in enumerate(self.langs)}
```

Comment:
We should probably add a check here to ensure that len(lang_probs) == len(datasets.keys()).
Also, would it be better to use a dictionary with the format <lang_id_str>: prob_lang_float instead of simply a list for lang_probs, this would make it more user friendly and also help avoid errors where there might be a mismatch in the order of languages in lang_probs and datasets?

Reply:
I have changed lang_probs to be a dictionary, and also added an assert check for length. Can you please check the new code?